src/cc/bpf_module.cc

```diff
@@ -134,7 +135,14 @@ BPFModule::~BPFModule() {
   engine_.reset();
   rw_engine_.reset();
   ctx_.reset();
+  if (do_trim) {
+    // return the freed top heap memory back to the system.
+    malloc_trim(0);
```

Comment:
> if we can always immediately free the execution engine once we are done using it.
One particular issue related to this is sscanf/snprintf functionality. User could still use it.
so rw_engine_ may never be able to freed without explicit permission from users.

Let me do an experiment first: only reset engine_, but leave rw_engine_ and llvm ctx_ to
see how much saving. If most of memory is in engine_ itself (for the case no actual sscanf/snprintf usage from user), I will go with suggested approach.

For the saving, malloc_trim is only used for glibc. I can hack to find the real saving. In our production system with jemalloc (which will do trimming automatically), the saving is more than 2MB for one moderate size bpf program.